src/cls/IPM/Utils/Module.cls

```diff
@@ -339,7 +339,15 @@ ClassMethod LoadModuleReference(pServerName As %String, pModuleName As %String,
 				Set tSC = ..LoadModuleFromArchive(tModRef.Name,tModRef.VersionString,tTmpStream,.pParams)
 			} Else {
 				// Old format (TODO: officially deprecate): try loading a .xml file
-				Set tSC = $System.OBJ.Load(tTmpStream.Filename,.qstruct)
+
+				// Load without compiling -> Update legacy superclasses -> Compile
+				If $Data(qstruct) \ 2 = 1 {
+					Set qstruct = $Translate(qstruct,"c","")
+				} ElseIf $Data(qstruct("/compile")) {
+					Kill qstruct("/compile")	
+				}
+				Set tSC = $System.OBJ.Load(tTmpStream.Filename,.qstruct, , .tLoaded)
+				Set tSC = $$$ADDSC(tSC, ##class(%IPM.Utils.LegacyCompat).UpdateSuperclassAndCompile(.tLoaded, .qstruct))
```

Comment:
If we are updating loading code in many different places, would it be worthwhile adding helper methods for loading and compiling classes? Then we have one place that would consistently manage loading and compiling in the package manager.

Reply:
Hi Keshav,

There aren't that many different places and we use different ways of importing/loading – $System.OBJ.ImportDir, $System.OBJ.Load. Sometimes we don't even load before compile.

If more repetitive code comes along, I'll add the new methods.